File: bmizad3/app/src/main/java/com/project/bmi2/Activity4.java
package com.project.bmi2;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;


public class Activity4 extends AppCompatActivity {
    private final String SCORE_KEY = "SCORE";
    private final String INDEX_KEY = "INDEX";
    private TextView mTxtQuestion;
    private Button btnF;
    private Button btnT;
    private int mQuestionIndex;
    private int mQuizQuestion;
    private ProgressBar mProgressBar;
    private TextView mQuizstatsTextView;
    private int mUserScore;


    private final QuizModel[] questionCollection = new QuizModel[]{
            new QuizModel(R.string.q1, false),
            new QuizModel(R.string.q2, true),
            new QuizModel(R.string.q3, true),
            new QuizModel(R.string.q4, true),
            new QuizModel(R.string.q5, true),
            new QuizModel(R.string.q6, true),



    };
    final int USER_PROGRESS = (int) Math.ceil(100.0 / questionCollection.length);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_4);
        mTxtQuestion = findViewById(R.id.txtQuestion);
        QuizModel q1 = questionCollection[mQuestionIndex];
        mQuizQuestion = q1.getmQuestion();
        mTxtQuestion.setText(mQuizQuestion);
        mProgressBar = findViewById(R.id.quizPB);
        mQuizstatsTextView = findViewById(R.id.txtQuizStats);


        btnT = findViewById(R.id.btnTrue);
        btnT.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                evaluateUserAnswer(true);
                changeQuestionOnButtonClick();

            }
        });

        btnF = findViewById(R.id.btnFalse);
        btnF.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                evaluateUserAnswer(false);
                changeQuestionOnButtonClick();

            }
        });

    }

    private void changeQuestionOnButtonClick() {
        mQuestionIndex = (mQuestionIndex + 1) % 6;
        if (mQuestionIndex == 0) {
            AlertDialog.Builder quizAlert = new AlertDialog.Builder(this);
            quizAlert.setCancelable(false);
            quizAlert.setTitle("Quiz is Finished");
            quizAlert.setMessage("Your Score is " + mUserScore);
            quizAlert.setPositiveButton("Finish the Quiz", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                    finish();
                }
            });
            quizAlert.show();
        }
        mQuizQuestion = questionCollection[mQuestionIndex].getmQuestion();
        mTxtQuestion.setText(mQuizQuestion);
        mProgressBar.incrementProgressBy(USER_PROGRESS);
        mQuizstatsTextView.setText(mUserScore + "" + "/ 6");
    }

    private void evaluateUserAnswer(boolean userGuess) {
        boolean currentQuestionAnswer = questionCollection[mQuestionIndex].ismAnswer();
        if (currentQuestionAnswer == userGuess) {

            mUserScore = mUserScore + 1;
        } else {
        }
    }

}

File: bmizad4/app/src/main/java/com/project/bmi2/Activity4.kt
package com.project.bmi2

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity

class Activity4 : AppCompatActivity() {
    private val SCORE_KEY = "SCORE"
    private val INDEX_KEY = "INDEX"
    private var mTxtQuestion: TextView? = null
    private var btnF: Button? = null
    private var btnT: Button? = null
    private var mQuestionIndex = 0
    private var mQuizQuestion = 0
    private var mProgressBar: ProgressBar? = null
    private var mQuizstatsTextView: TextView? = null
    private var mUserScore = 0
    private val questionCollection = arrayOf(
            QuizModel(R.string.q1, false),
            QuizModel(R.string.q2, true),
            QuizModel(R.string.q3, true),
            QuizModel(R.string.q4, true),
            QuizModel(R.string.q5, true),
            QuizModel(R.string.q6, true))
    val USER_PROGRESS = Math.ceil(100.0 / questionCollection.size).toInt()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_4)
        mTxtQuestion = findViewById(R.id.txtQuestion)
        val q1 = questionCollection[mQuestionIndex]
        mQuizQuestion = q1.getmQuestion()
        mTxtQuestion.setText(mQuizQuestion)
        mProgressBar = findViewById(R.id.quizPB)
        mQuizstatsTextView = findViewById(R.id.txtQuizStats)
        btnT = findViewById(R.id.btnTrue)
        btnT.setOnClickListener(View.OnClickListener {
            evaluateUserAnswer(true)
            changeQuestionOnButtonClick()
        })
        btnF = findViewById(R.id.btnFalse)
        btnF.setOnClickListener(View.OnClickListener {
            evaluateUserAnswer(false)
            changeQuestionOnButtonClick()
        })
    }

    private fun changeQuestionOnButtonClick() {
        mQuestionIndex = (mQuestionIndex + 1) % 6
        if (mQuestionIndex == 0) {
            val quizAlert = AlertDialog.Builder(this)
            quizAlert.setCancelable(false)
            quizAlert.setTitle("Quiz is Finished")
            quizAlert.setMessage("Your Score is $mUserScore")
            quizAlert.setPositiveButton("Finish the Quiz") { dialogInterface, i -> finish() }
            quizAlert.show()
        }
        mQuizQuestion = questionCollection[mQuestionIndex].getmQuestion()
        mTxtQuestion!!.setText(mQuizQuestion)
        mProgressBar!!.incrementProgressBy(USER_PROGRESS)
        mQuizstatsTextView!!.text = "$mUserScore/ 6"
    }

    private fun evaluateUserAnswer(userGuess: Boolean) {
        val currentQuestionAnswer = questionCollection[mQuestionIndex].ismAnswer()
        if (currentQuestionAnswer == userGuess) {
            mUserScore = mUserScore + 1
        } else {
        }
    }
}